### ISSUE TITLE
Training on 2 Machines (16 GPUs) is Slower than 1 Machine (8 GPUs)

### ISSUE THREAD

**Issue by mingkai-zheng** [rhymes-ai/Aria#101]
Hi, I'm fine-tuning the Aria Model using DeepSpeed ZeRO Stage 3 and gradient checkpointing. However, I’ve encountered a significant performance issue - Training on 2 machines (16 GPUs in total) is much slower than training on a single machine (8 GPUs). The issue seems specific to the Aria Model—when I test with other models like Phi, I do not observe the same slowdown. 

Have you encountered similar multi-machine slowdowns with DeepSpeed ZeRO-3?
Are there any specific DeepSpeed configurations or network tuning parameters that might help?
Could this be due to an issue with how ZeRO-3 shards the model state across machines for the Aria model?

**Comment by xffxff** [comment]
Could you share your deepseed config?

Also, did you install `grouped-gemm`, as it may significantly affect training throughput

**Comment by mingkai-zheng** [comment]
I installed grouped-gemm.

I launched DeepSpeed directly using the accelerate CLI with the following configuration:
```
accelerate launch  --use_deepspeed  --zero_stage 3 \
     --offload_optimizer_device none  --offload_param_device none  --zero3_init_flag true \
     --zero3_save_16bit_model true  --deepspeed_exclusion_filter None  --deepspeed_inclusion_filter None  \
     --num_processes 16 --num_machines 2 --machine_rank 0 --deepspeed_multinode_launcher standard  ....
```

The accelerate provides a detailed log of the DeepSpeed configuration, as shown below:

```
[INFO] optimizer_legacy_fusion ...... False
[INFO] optimizer_name ............... None
[INFO] optimizer_params ............. None
[INFO] pipeline ..................... {'stages': 'auto', 'partition': 'best', 'seed_layers': False, 'activation_checkpoint_interval': 0, 'pipe_partitioned': True, 'grad_partitioned': True}
[INFO] pld_enabled .................. False
[INFO] pld_params ................... False
[INFO] prescale_gradients ........... False
[INFO] scheduler_name ............... None
[INFO] scheduler_params ............. None
[INFO] seq_parallel_communication_data_type  torch.float32
[INFO] sparse_attention ............. None
[INFO] sparse_gradients_enabled ..... False
[INFO] steps_per_print .............. inf
[INFO] timers_config ................ enabled=True synchronized=True
[INFO] train_batch_size ............. 256
[INFO] train_micro_batch_size_per_gpu  1
[INFO] use_data_before_expert_parallel_  False
[INFO] use_node_local_storage ....... False
[INFO] wall_clock_breakdown ......... False
[INFO] weight_quantization_config ... None
[INFO] world_size ................... 16
[INFO] zero_allow_untested_optimizer  True
[INFO] zero_config .................. stage=3 contiguous_gradients=True reduce_scatter=True reduce_bucket_size=500000000 use_multi_rank_bucket_allreduce=True allgather_partitions=True allgather_bucket_size=500000000 overlap_comm=True load_from_fp32_weights=True elastic_checkpoint=False offload_param=DeepSpeedZeroOffloadParamConfig(device='none', nvme_path=None, buffer_count=5, buffer_size=100000000, max_in_cpu=1000000000, pin_memory=False) offload_optimizer=DeepSpeedZeroOffloadOptimizerConfig(device='none', nvme_path=None, buffer_count=4, pin_memory=False, pipeline_read=False, pipeline_write=False, fast_init=False, ratio=1.0) sub_group_size=1000000000 cpu_offload_param=None cpu_offload_use_pin_memory=None cpu_offload=None prefetch_bucket_size=50000000 param_persistence_threshold=100000 model_persistence_threshold=9223372036854775807 max_live_parameters=1000000000 max_reuse_distance=1000000000 gather_16bit_weights_on_model_save=True use_all_reduce_for_fetch_params=False stage3_gather_fp16_weights_on_model_save=False ignore_unused_parameters=True legacy_stage1=False round_robin_gradients=False zero_hpz_partition_size=1 zero_quantized_weights=False zero_quantized_nontrainable_weights=False zero_quantized_gradients=False mics_shard_size=-1 mics_hierarchical_params_gather=False memory_efficient_linear=True pipeline_loading_checkpoint=False override_module_apply=True
[INFO] zero_enabled ................. True
[INFO] zero_force_ds_cpu_optimizer .. True
[INFO] zero_optimization_stage ...... 3
[INFO] json = {
    "train_batch_size": 256, 
    "train_micro_batch_size_per_gpu": 1, 
    "gradient_accumulation_steps": 16, 
    "zero_optimization": {
        "stage": 3, 
        "offload_optimizer": {
            "device": "none", 
            "nvme_path": null
        }, 
        "offload_param": {
            "device": "none", 
            "nvme_path": null
        }, 
        "stage3_gather_16bit_weights_on_model_save": true
    }, 
    "gradient_clipping": 1.0, 
    "steps_per_print": inf, 
    "bf16": {
        "enabled": true
    }, 
    "fp16": {
        "enabled": false
    }, 
    "zero_allow_untested_optimizer": true
}
```
